File: UnitTesting/SpecFlow-Examples-master/BowlingKata/BowlingKata-Fsharp/Bowling.SpecFlow/StepDefinitions/BowlingSteps.fs
module Bowling.SpecFlow.StepDefinitions

open Bowling.SpecFlow.Drivers
open TechTalk.SpecFlow

[<Binding>]
type public BowlingSteps(driver : BowlingDriver) =
    let _driver : BowlingDriver = driver
    let [<Given>]``a new bowling game``() =
       _driver.NewGame()

    let [<When>]``all of my balls are landing in the gutter``() =
        _driver.Throw 20 0

    let [<When>]``all of my rolls are strikes``() =
        _driver.Throw 12 10
     
    let [<Then>]``my total score should be (\d+)`` score =
        _driver.CheckScore score

    let [<When>]``I roll (\d+)`` pins =
         _driver.Throw 1 pins
     
    let [<When>]``I roll (\d+) and (\d+)`` pins1 pins2 =
        _driver.DoubleThrow 1 pins1 pins2
     
    let [<When>]``I roll (\d+) times (\d+) and (\d+)`` count pins1 pins2 =
        _driver.DoubleThrow count pins1 pins2
     
    let [<When>]``I roll the following series:(.*)`` (series:string) =
        _driver.ThrowSeriesFromString series
     
    let [<When>]``I roll`` (rolls:Table) =
        _driver.ThrowSeriesFromTable rolls

File: UnitTesting/Old/SpecFlow-Examples-master/BowlingKata/BowlingKata-German/Bowling.Specflow/StepDefinitions/BowlingSteps.cs
using System;
using Bowling.Specflow.Drivers;
using TechTalk.SpecFlow;

namespace Bowling.Specflow.StepDefinitions
{
    [Binding]
    public class BowlingSteps
    {
        private readonly BowlingDriver _driver;

        public BowlingSteps(BowlingDriver driver)
        {
            _driver = driver ?? throw new ArgumentNullException(nameof(driver));
        }

        [Given(@"eine neue Bowling-Partie")]
        public void GivenANewBowlingGame()
        {
            _driver.NewGame();
        }

        [When(@"alle meine Kugeln in der Seitenrinne landen")]
        public void WhenAllOfMyBallsAreLandingInTheGutter()
        {
            _driver.Roll(0, 20);
        }

        [When(@"ich nur Strikes werfe")]
        public void WhenAllOfMyRollsAreStrikes()
        {
            _driver.Roll(10, 12);
        }

        [Then(@"soll meine Punktzahl (\d+) sein")]
        public void ThenMyTotalScoreShouldBe(int score)
        {
            _driver.CheckScore(score);
        }

        [When(@"ich (\d+) werfe")]
        public void WhenIRoll(int pins)
        {
            _driver.Roll(pins, 1);
        }

        [When(@"ich (\d+) und (\d+) werfe")]
        public void WhenIRoll(int pins1, int pins2)
        {
            _driver.Roll(pins1, pins2, 1);
        }

        [When(@"ich (\d+) mal (\d+) und (\d+) werfe")]
        public void WhenIRollSeveralTimes2(int rollCount, int pins1, int pins2)
        {
            _driver.Roll(pins1, pins2, rollCount);
        }

        [When(@"ich folgende Serie werfe:(.*)")]
        public void WhenIRollTheFollowingSeries(string series)
        {
            _driver.RollSeries(series);
        }

        [When(@"ich werfe")]
        public void WhenIRoll(Table rolls)
        {
            _driver.RollSeries(rolls);
        }
    }
}
